src/dswx_sar/dswx_runconfig.py

```diff
@@ -289,15 +347,22 @@ def load_from_yaml(cls, yaml_path: str, workflow_name: str, args):
 
         algorithm_cfg = load_validate_yaml(ancillary.algorithm_parameters,
                                            f'algorithm_parameter_{sensor.lower()}')
+
+        # Check if input files have the requested polarizations and
+        # sort the order of the polarizations.
+        input_dir_list = \
+            cfg['runconfig']['groups']['input_file_group']['input_file_path']
+        requested_pol = algorithm_cfg['runconfig']['processing']['polarizations']
         co_pol, cross_pol, pol_list = check_polarizations(
-            algorithm_cfg['runconfig']['processing']['polarizations'])
-        
+            requested_pol, input_dir_list)
+
+        # update the polarizations
         algorithm_cfg['runconfig']['processing']['polarizations'] = pol_list
-        algorithm_cfg['runconfig']['processing']['copol'] = co_pol[0]
-        algorithm_cfg['runconfig']['processing']['crosspol'] = cross_pol[0]
+        algorithm_cfg['runconfig']['processing']['copol'] = co_pol[0] if co_pol else None
+        algorithm_cfg['runconfig']['processing']['crosspol'] = cross_pol[0] if cross_pol else None
```

Comment:
What is going to happen if there is a two different dual-pols in the burst RTC list?
i.e. `['VV', 'VH']` and `['HH', 'HV']`?

Reply:
I think this question is resolved now.  For example, if the bursts have four polarizations, they are going to be processed separately. 
In the above lines, `['VV']` and `['HH']` are assigned to the `copol`.